plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/storage/TestTrinoHudiStorage.java

```diff
@@ -0,0 +1,294 @@
+        storage.deleteFile(path);
+
+        byte[] data = new byte[] {2, 42, 49, (byte) 158, (byte) 233, 66, 9};
+
+        try (OutputStream stream = storage.create(path)) {
+            stream.write(data);
+            stream.flush();
+        }
+        validatePathInfo(storage, path, data, false);
+
+        assertThatThrownBy(() -> storage.create(path, false))
+                .isInstanceOf(IOException.class);
+        validatePathInfo(storage, path, data, false);
+
+        assertThatThrownBy(() -> storage.create(path, false))
+                .isInstanceOf(IOException.class);
+        validatePathInfo(storage, path, data, false);
+
+        StoragePath path2 = new StoragePath(getTempDir(), "testCreateAppendAndRead/2.file");
+        assertThat(storage.exists(path2)).isFalse();
+        assertThat(storage.createNewFile(path2)).isTrue();
+        validatePathInfo(storage, path2, EMPTY_BYTES, false);
+        assertThat(storage.createNewFile(path2)).isFalse();
+
+        StoragePath path3 = new StoragePath(getTempDir(), "testCreateAppendAndRead/3.file");
+        assertThat(storage.exists(path3)).isFalse();
+        storage.createImmutableFileInPath(path3, Option.of(data));
+        validatePathInfo(storage, path3, data, false);
+
+        StoragePath path4 = new StoragePath(getTempDir(), "testCreateAppendAndRead/4");
+        assertThat(storage.exists(path4)).isFalse();
+        assertThat(storage.createDirectory(path4)).isTrue();
+        validatePathInfo(storage, path4, EMPTY_BYTES, true);
+        assertThat(storage.createDirectory(path4)).isTrue();
+    }
+
+    private HoodieStorage getStorage()
+    {
+        Object config = getConf();
+        return getStorage(getFileSystem(config), config);
+    }
+
+    private StoragePathInfo getStoragePathInfo(String subPath, boolean isDirectory)
+    {
+        return new StoragePathInfo(new StoragePath(getTempDir(), subPath),
+                0, isDirectory, (short) 1, 1000000L, 10L);
+    }
+
+    private static void validatePathInfo(
+            HoodieStorage storage,
+            StoragePath path,
+            byte[] data,
+            boolean isDirectory)
+            throws IOException
+    {
+        assertThat(storage.exists(path)).isTrue();
+        StoragePathInfo pathInfo = storage.getPathInfo(path);
+        assertThat(pathInfo.getPath()).isEqualTo(path);
+        assertThat(pathInfo.isDirectory()).isEqualTo(isDirectory);
+        assertThat(pathInfo.isFile()).isEqualTo(!isDirectory);
+        if (!isDirectory) {
+            assertThat(pathInfo.getLength()).isEqualTo(data.length);
+            try (InputStream stream = storage.open(path)) {
+                assertThat(IOUtils.readAsByteArray(stream, data.length)).isEqualTo(data);
+            }
+            assertThat(pathInfo.getModificationTime()).isGreaterThan(0);
+        }
+    }
+
+    private static void validatePathInfoList(
+            List<StoragePathInfo> expected,
+            List<StoragePathInfo> actual)
```

Comment:
Replace the order of the parameters for consistency with AssertJ assertions.

Reply:
Changed to `actual` as the first argument.